src/main/webapp/components/directives/cache/security/security.js

```diff
@@ -0,0 +1,158 @@
+(function () {
+  'use strict';
+
+  var module = angular.module('ispn.directives.cache.security', ['ispn.services.utils']);
+
+  module.directive('security', ['utils', '$modal', function (utils, modal) {
+    return {
+      restrict: 'E',
+      scope: {
+        data: '=',
+        cluster: '=',
+        metadata: '=',
+        initDefaults: '=',
+        readOnly: '=',
+        outsideController: '='
+      },
+      replace: true,
+      templateUrl: 'components/directives/cache/security/security.html',
+      link: function (scope) {
+        if (utils.isNotNullOrUndefined(scope.outsideController)) {
+          if (utils.isArray(scope.outsideController)) {
+            var handle = {};
+            scope.outsideController.push(handle);
+            scope.internalController = handle;
+          } else {
+            scope.internalController = scope.outsideController;
+          }
+        } else {
+          scope.internalController = {};
+        }
+        scope.hasSecurityBeenDefinedGlobally = function () {
+          return utils.isNotNullOrUndefined(scope.cluster.security) && utils.isNotNullOrUndefined(scope.cluster.security.SECURITY.authorization.AUTHORIZATION) ;
+        };
+
+        if (utils.isNullOrUndefined(scope.data)) {
+          scope.data = {};
+          utils.deepSet(scope.data, "SECURITY.authorization.AUTHORIZATION", {});
+          scope.data = scope.data.SECURITY.authorization.AUTHORIZATION;
```

Comment:
This does not work as expected when we pass the data through to the `cache-creation-controller.service` as we have not set the 'is-new-node' field to true on either `security.SECURITY` or `security.SECURITY.authorization.AUTHORIZATION` nodes.  This results in the following error when we attempt to update an existing config with security.SECURITY undefined:

```WFLYCTL0216: Management resource '[ ("profile" => "clustered"), ("subsystem" => "datagrid-infinispan"), ("cache-container" => "clustered"), ("configurations" => "CONFIGURATIONS"), ("distributed-cache-configuration" => "default"), ("security" => "SECURITY") ]' not found```